beacon-chain/rpc/eth/shared/conversions_state.go

```diff
@@ -0,0 +1,592 @@
+package shared
+
+import (
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	beaconState "github.com/prysmaticlabs/prysm/v4/beacon-chain/state"
+	enginev1 "github.com/prysmaticlabs/prysm/v4/proto/engine/v1"
+)
+
+func BeaconStateFromConsensus(st beaconState.BeaconState) (*BeaconState, error) {
+	srcbr := st.BlockRoots()
+	br := make([]string, len(srcbr))
+	for i, r := range srcbr {
+		br[i] = hexutil.Encode(r)
+	}
+	srcsr := st.StateRoots()
+	sr := make([]string, len(srcsr))
+	for i, r := range srcsr {
+		sr[i] = hexutil.Encode(r)
+	}
+	srchr, err := st.HistoricalRoots()
+	if err != nil {
+		return nil, err
+	}
+	hr := make([]string, len(srchr))
+	for i, r := range srchr {
+		hr[i] = hexutil.Encode(r)
+	}
+	srcvotes := st.Eth1DataVotes()
+	votes := make([]*Eth1Data, len(srcvotes))
+	for i, e := range srcvotes {
+		votes[i] = Eth1DataFromConsensus(e)
+	}
+	srcvals := st.Validators()
+	vals := make([]*Validator, len(srcvals))
+	for i, v := range srcvals {
+		vals[i] = ValidatorFromConsensus(v)
+	}
+	srcbals := st.Balances()
+	bals := make([]string, len(srcbals))
+	for i, b := range srcbals {
+		bals[i] = fmt.Sprintf("%d", b)
+	}
+	srcrm := st.RandaoMixes()
+	rm := make([]string, len(srcrm))
+	for i, m := range srcrm {
+		rm[i] = hexutil.Encode(m)
+	}
+	srcslashings := st.Slashings()
+	slashings := make([]string, len(srcslashings))
+	for i, s := range srcslashings {
+		slashings[i] = fmt.Sprintf("%d", s)
+	}
+	srcPrevAtts, err := st.PreviousEpochAttestations()
```

Comment:
This is a bit of a nit, but some of these are camel case while others are all lower case